classes/Config.js

```diff
@@ -141,15 +147,18 @@ class CollectionConfig extends Config {
 
   async deleteItemFromOrder(item) {
     const collectionName = this.collectionName
-
-    const { content, sha } = await this.read()
-    const contentObject = yaml.safeLoad(base64.decode(content))
-    
+    const { contentObject, sha } = await this.read()
     const index = contentObject.collections[collectionName].order.indexOf(item);
     contentObject.collections[collectionName].order.splice(index, 1)
     const newContent = base64.encode(yaml.safeDump(contentObject))
     
     await this.update(newContent, sha)
+    return { index, item }
+  }
+
+  async updateItemInOrder(oldItem, newItem) {
```

Comment:
It's nice to compose the `updateItemInOrder` from the other methods, but this results in two API calls for what is essentially one update. Have we considered writing specific "update" code so only one API call is made?

Reply:
@gweiying brought up the possibility of building out the methods for the `CollectionConfig` class but as that would be a more significant undertaking we will be postponing that for now. As discussed, we will modify the `updateItemInOrder` method so that the the directory config file is read only once, and custom updating logic is applied instead of composing it with the `add` and `delete` methods

Reply:
e4f835a